moskito-backend/src/Controller/UserController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\UserRepository;
+use App\Entity\User;
+use App\Serializer\UserSerializer;
+
+class UserController extends AbstractController
+{
+    /**
+     * @Route("/user", methods={"GET"})
+     */
+    public function index( UserRepository $userRepository, UserSerializer $serializer ): JsonResponse
+    {
+        $users = $userRepository->findAll();
+
+
+        return new JsonResponse(
+            $serializer->serialize($users),
+            JsonResponse::HTTP_OK,
+            [],
+            true
+        );
+    }
+
+
+    /**
+     * @Route("/register", methods={"POST"})
+     */
+    public function register(
+        Request $request, 
+        UserRepository $userRepository, 
+        UserSerializer $serializer
+        ): JsonResponse {
+    
+            $user = $serializer->deserialize($request->getContent());
+            $allUsers = $userRepository->findAll();
```

Comment:
You could use findBy to look for the submitted email address in your db. If a match is found send the error response accordingly. This would have the advantage of not having to loop over all db entries.

Reply:
good point! changed that!